e2e-tests/e2eTest.spec.ts

```diff
@@ -125,9 +158,53 @@ void describe('e2e-tests', () => {
 		assert.equal(req.headers.get('content-length'), '0')
 		assert.equal(await req.text(), '')
 	})
+	const expectedResLastHour: Array<Record<string, string | number>> = []
+	timestampsLastHour.forEach((ts, index) => {
+		expectedResLastHour.push({
+			ts: ts.toISOString(),
+			usedBytes: randomVal[index] ?? 0,
+		})
+	})
+	const expectedResLastHourOnomondo: Array<Record<string, string | number>> = []
+	timestampsLastHour.forEach((ts, index) => {
+		expectedResLastHourOnomondo.push({
+			ts: ts.toISOString(),
+			usedBytes: randomVal2[index] ?? 0,
+		})
+	})
+	const expectedResLastDay: Array<Record<string, string | number>> = []
+	timestampsLastDay.forEach((ts, index) => {
+		expectedResLastDay.push({
+			ts: ts.toISOString(),
+			usedBytes: randomVal[index] ?? 0,
+		})
+	})
+	for (const [iccid, response, timespan] of [
+		[iccidNewWL, expectedResLastHour, 'lastHour'],
+		[iccidOldWL, expectedResLastDay, 'lastDay'],
+		[iccidNew, expectedResLastHourOnomondo, 'lastHour'],
+	] as [string, Array<{ ts: string; usedBytes: number }>, string][]) {
+		void it(`should return measurements from timespan ${timespan} for iccid ${iccid}`, async () => {
+			const req = await fetchHistoricalData(iccid, timespan)
+			const expectedCacheControl = 'public, max-age=300'
+			const responseBody = await req.json()
+			assert.equal(req.headers.get('cache-control'), expectedCacheControl)
+			assert.equal(req.headers.get('Access-Control-Allow-Origin'), '*')
+			assert.equal(req.status, 200)
+			assert.deepEqual(responseBody, { measurements: response.reverse() })
+		})
+	}
 })
 
 const fetchData = async (iccid: string): Promise<Response> => {
 	const url = `${outputs.APIURL}/sim/${iccid}`
 	return await fetch(url)
 }
+
+const fetchHistoricalData = async (
+	iccid: string,
+	timeSpan: string,
+): Promise<Response> => {
+	const url = `${outputs.APIURL}/sim/${iccid}/historicalData/${timeSpan}`
+	return await fetch(url)
+}
```

Comment:
Do not access globals in a function, makes it harder to move it to a new file.

Should be refactored to
 
```suggestion
const fetchHistoricalData = (APIURL: URL) => async (
	iccid: string,
	timeSpan: string,
) => fetch(`${APIURL.toString()}/sim/${iccid}/historicalData/${timeSpan}`)
```

This code also ensures that APIURL is actually a URL.

**Remember:** Always use proper value objects (e.g. Date, or URL instead of strings).